ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1194,6 +1194,34 @@ void GCS_MAVLINK_Copter::handleMessage(mavlink_message_t* msg)
             break;
 #endif
 
+        case MAV_CMD_DO_WINCH:
+            // param1 : winch number (ignored)
+            // param2 : action (0=release, 1=length control, 2=rate control). See WINCH_ACTIONS enum.
+            if(!copter.g2.winch.enabled()) {
+                result = MAV_RESULT_FAILED;
+            } else {
+                result = MAV_RESULT_ACCEPTED;
+                switch ((uint8_t)packet.param2) {
+                    case WINCH_DISABLE:
+                        copter.g2.winch.relax();
+                        break;
+                    case WINCH_LENGTH_CONTROL: {
+                        float desired_length = copter.g2.winch.get_line_length() + packet.param3;
```

Comment:
Adding the current line length here is incorrect. You already do that inside ```AP_Winch::release_length()``` which is effectively increasing/decreasing the amount of length actually commanded. And inconsistent with how the command is used in a mission.

Reply:
great, GCS_MAVLink control fixed in follow-up commit.